airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -94,33 +98,41 @@ def _get_dep_statuses(
         session: Session,
         dep_context: DepContext,
     ) -> Iterator[TIDepStatus]:
+        # get all the setup tasks upstream of this task
+        setup_upstream_tasks = list(ti.task.get_upstreams_only_setups())
         # Checking that all upstream dependencies have succeeded.
         if not ti.task.upstream_task_ids:
             yield self._passing_status(reason="The task instance did not have any upstream tasks.")
             return
-        if ti.task.trigger_rule == TR.ALWAYS:
+        if ti.task.trigger_rule == TR.ALWAYS and not setup_upstream_tasks:
+            # even with ALWAYS trigger rule, we still need to check setup tasks
```

Comment:
i think always should probably keep the same behavior.  it says in the docs `No dependencies at all, run this task at any time`.  it used to be called "dummy" trigger rule.  so i think we just keep the absolute short circuit here.

Reply:
it's really a non-sensical rule. with it, the deps are just for show. no one should ever really use this trigger rule.  perhaps it's there for testing 🤷